### PR TITLE
Add a way to record requests and responses - Fixes #80

### LINKED ISSUE #80: Allow recording request/response stats into CSV file
Hi,
Would you consider writing raw data into simple text format, for further analysis by other scripts? Something like JMeter/Grinder/Gatling already do, where one request-response pair has one record in CSV with sufficient fields like response time, time to first byte, request label, bytes counts etc.

That would allow to plug Molotov into other solutions which use this approach of writing file and analysing it after test or in parallel. Namely Taurus Tool or Jenkins Performance Plugin.

### PR DESCRIPTION
**Pull request by tarekziade**
- [x] Make the session request response printer pluggable
- [x] add a "events" fixture so people can write their own backends
- [x] document the feature with an example
- [x] add a CLI option. --use-extension + tests

### CODE REVIEW

File: molotov/listeners.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+_UNREADABLE = "***WARNING: Molotov can't display this body***"
+_BINARY = "**** Binary content ****"
+_FILE = "**** File content ****"
+_COMPRESSED = ('gzip', 'compress', 'deflate', 'identity', 'br')
+
+
+class BaseListener(object):
+    async def __call__(self, session, event, **options):
+        attr = getattr(self, 'on_' + event)
+        await attr(**options)
```
Comment (peterbe) [inline molotov/listeners.py]: Could this instead be something like 
```python
attr = getattr(self, 'on_' + event, None)
if attr is not None:
    await attr(**options)
```
That way, if I writer a listener class (based on `BaseListener`) and I'm *only* interested in `on_response_received` I wouldn't *have to* write...
```python
    async def on_receive_request(self, *i, **dontcare):
            pass
```

File: molotov/tests/example4.py [hunk 2]
```diff
@@ -11,6 +11,11 @@ def cleanup(self):
         pass
 
 
+@molotov.session_events()
+async def event(session, event, **data):
+    print(event)
+
```
Comment (peterbe) [inline molotov/tests/example4.py]: Nice!

File: molotov/listeners.py (hunk 1 above)
Comment (tarekziade) [reply molotov/listeners.py]: Good idea

**Comment by tarekziade** [comment]
I came up with a better name and mechanism. --use-extension /path/to/a/python/module.py
(can be used multiple times)

This will simply import the module, so you can have a session_events() in its own utility module and use it against a separate load test module. or your own custom setup() etc..

**Comment by undera** [comment]
Sounds great! Thanks!

**Pull request merged**